### PR TITLE
📕 Add hook guides

### PR DESCRIPTION
**Pull request by nezouse**

### CODE REVIEW

File: docs/source/guide.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ The hook will retrieve a balance of an ERC20 token of the provided address.
 
 .. code-block:: javascript
 
-  function useTokenBalance(address: string | Falsy, tokenAddress: string | Falsy) {
+  function useTokenBalance(tokenAddress: string | Falsy, address: string | Falsy) {
```
Comment (rzadp) [inline docs/source/guide.rst:32]: Shouldn't this change be in https://github.com/EthWorks/useDApp/pull/85 ?

Comment (nezouse) [reply docs/source/guide.rst:32]: Sure, I forgot about this file when I was making the changes.

File: docs/source/guide.rst [hunk 2]
```diff
@@ -39,14 +39,136 @@ The hook will retrieve a balance of an ERC20 token of the provided address.
...
+        }
+      )
+      await waitForCurrent((val) => val !== undefined)
+
+      expect(result.error).to.be.undefined
+      expect(result.current).to.eq(utils.parseEther('1'))
+    })
+  })
+
+First, we create a mock provider for our token. We save it to the variable as we need it later.
```
Comment (marekkirejczyk) [inline docs/source/guide.rst:120]: We save it to the variable as we need it later. -> Unnecessary

File: docs/source/guide.rst [hunk 3]
```diff
@@ -39,14 +39,136 @@ The hook will retrieve a balance of an ERC20 token of the provided address.
...
+Rules of hooks apply.
+
 All core hooks are available `here <https://github.com/EthWorks/useDapp/tree/master/packages/core/src/hooks>`_.
 
 
 Testing hooks
 *************
 
-Coming soon...
+Testing should be easy and straightforward. We just need a small setup and we are good to go.
```
Comment (marekkirejczyk) [inline docs/source/guide.rst:78]: Opinion. Focus on facts. Better:
To write test, start with a setup code that will create a mock provider and  test wallets.

File: docs/source/guide.rst [hunk 4]
```diff
@@ -39,14 +39,136 @@ The hook will retrieve a balance of an ERC20 token of the provided address.
...
+        {
+          mockProvider,
+        }
+      )
+      await waitForCurrent((val) => val !== undefined)
+
+      expect(result.error).to.be.undefined
+      expect(result.current).to.eq(utils.parseEther('1'))
+    })
+  })
```
Comment (marekkirejczyk) [inline docs/source/guide.rst:118]: Add complete example at the end

File: docs/source/guide.rst [hunk 5]
```diff
@@ -39,14 +39,136 @@ The hook will retrieve a balance of an ERC20 token of the provided address.
...
+
+First, we create a mock provider for our token. We save it to the variable as we need it later.
+Then we get our wallets from the provider. We will need them to check if our hook is working correctly.
+
+.. code-block:: javascript
+
+  const mockProvider = new MockProvider()
+  const [deployer, spender] = mockProvider.getWallets()
+
+To end our setup before each test we deploy a fresh contract. It's important as otherwise the result of one
```
Comment (marekkirejczyk) [inline docs/source/guide.rst:128]: Before each test deploy a ERC20 contract.

File: docs/source/guide.rst [hunk 6]
```diff
@@ -39,14 +39,128 @@ The hook will retrieve a balance of an ERC20 token of the provided address.
     return tokenBalance
   }
 
+Another example can be useTokenAllowance. Instead of balanceOf we use allowance on ERC20 interface.
```
Comment (marekkirejczyk) [inline docs/source/guide.rst:42]: Another example is useTokenAllowance hook.

File: docs/source/guide.rst [hunk 7]
```diff
@@ -39,14 +39,128 @@ The hook will retrieve a balance of an ERC20 token of the provided address.
...
+
+  expect(result.error).to.be.undefined
+  expect(result.current).to.eq(utils.parseEther('1'))
+
+To check if the hook reads data correctly we need to prepare it first. We approve the spender so then we can check that our hook returns the correct value.
+
+To test the hook we need to render it using renderWeb3Hook. It works like `renderHook` from react-testing library,
+but it wraps the hook into additional providers.
+
+On blockchain everything is async. To make tests predictable we need to wait for the result.
```
Comment (marekkirejczyk) [inline docs/source/guide.rst:121]: With react components updates happens asynchronically, as well as getting updates form blockchain. To get the first value, wait for value to be set.

**Review by marekkirejczyk — APPROVED** [review]

**Pull request merged**